docker/knora-api/BUILD.bazel

```diff
@@ -0,0 +1,38 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_docker//container:container.bzl", "container_image", "container_push")
+
+# builds the docker image
+container_image(
+    name = "image",
+    base = "@openjdk11//image",
```

Comment:
I see that you have added the build of docker images into Bazel and have removed the corresponding docker-compose of every single image? I feel like it makes the workflow harder to understand. What do you see as an advantage here?

Reply:
One of the problems we now have is the disconnect between `docker build` (or any of the make targets) and `sbt compile`. You only want to build a new image if the source code has changed or any of the inputs to the docker image. For this, we need to build Docker images with Bazel. With Bazel, we can generate a nice graph where all the dependencies are depicted. This should help to understand how things are connected.

The current plumbing with Docker and SBT was more of a hack than a nice implementation. SBT has its own way of defining Docker images inside SBT, but I had great problems understanding how to do it. In general, everything we now have with Bazel should/needed to be also be done in a similar way completely with SBT. The only difference is that it took me way less time to do these things in Bazel. I tried many times to do the same in SBT over the past 5 years, but I simply don't understand SBT. To complicated for me ;-)